packages/block-editor/src/components/provider/use-block-sync.js

```diff
@@ -113,6 +120,37 @@ export default function useBlockSync( {
 		onChangeRef.current = onChange;
 	}, [ onInput, onChange ] );
 
+	// Determine if blocks need to be reset when they change.
```

Comment:
The changes after this line are not necessary (just reordering the hooks). I believe this order makes more sense. "External" then "Internal" changes.

Reply:
Actually it seems the reordering has an effect, without it, this PR doesn't work :)